src/Documentation/Event-Sourcing/LogConsistencyProviders.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: page
+title: Included Log-Consistency Providers
+---
+
+# Built-In Log-Consistency Providers
+
+The `Microsoft.Orleans.EventSourcing` package includes several log-consistency providers that cover basic scenarios suitable to get started, and allow some extensibility.
+
+
+### Orleans.EventSourcing.**StateStorage**.LogConsistencyProvider
+
+This provider stores *grain state snapshots*, using a standard storage provider that can be configured independently. 
+
+The data that is kept in storage is an object that contains both the grain state (as specified by the first type parameter to `JournaledGrain`) and some meta-data (the version number, amd a special tag that is used to avoid duplication of events when storage accesses fail).
+
+Since the entire grain state is read/written every time we access storage, this provider is not suitable for objects whose grain state is very large. 
+
+This provider does not support `RetrieveConfirmedEvents`: it cannot retrieve the events from storage because the events are not persisted.
+
+### Orleans.EventSourcing.**LogStorage**.LogConsistencyProvider
+
+This provider stores *the complete event sequence as a single object*, using a standard storage provider that can be configured independently.
+
+The data that is kept in storage is an object that contains a `List<EventType> object`, and some meta-data (a special tag that is used to avoid duplication of events when storage accesses fail).
+
+This provider does support `RetrieveConfirmedEvents`. All events are always available and kept in memory.
```

Comment:
> The entire List is kept in memory because every time we read/write to/from storage, we have to read/write the entire list.

The problem here is that you are using our traditional Grain State storage to store all events and keeping then in memory. This will definitively don't scale. What we need is to persist into a real Event Store (like GES or StreamStone), read the stream when activating, build the state project and don't keep the event list in memory. Whenever people need to read events (i.e. to replay/republish them) you provide a give API like `GetEvents(start, end)` do whatever they need, and discard it. Only the grain state is kept in memory as a snapshot of all events applied.